### PR TITLE
Money printer: new role-restricted item for Cargo Techs and Quartermaster

### PR DESCRIPTION
**Pull request by YouCanBeBetterThanThat**
# Document the changes in your pull request

New Traitor role-restricted item for Cargo Techs and the QM: A printer that spits out 5000 credits every 120 seconds. Very obvious and very illegal.

Sprite:
image

# Wiki Documentation

Add to syndicate items page: 
	name = "Money Printer"
	desc = "A gigantic device that takes in nothing and puts out money. Sounds like a scam if you ask me. Don't get raided!"

# Changelog

:cl:  
rscadd: Money printer: new role-restricted item for Cargo Techs and Quartermaster
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
darkrp

**Comment by YouCanBeBetterThanThat** [comment]
> darkrp

rdm rdm rdm rdm rdm rdm rdm rdm

**Comment by KillerOrcaCora** [comment]
IF YOU COME INTO THIS ROOM IM SHOOTING MY RPG!!!

**Comment by YouCanBeBetterThanThat** [comment]
kos past line

**Review by nesisknesis — CHANGES_REQUESTED** [review]
image

You can't just downsize an image. I'll see if someone can make you an actual sprite

**Comment by YouCanBeBetterThanThat** [comment]
thanks!

File: yogstation/code/game/objects/items/money_printer.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	addtimer(CALLBACK(src, .proc/print), cooldown_per_cycle)
+
+/obj/item/money_printer/proc/print()
+	addtimer(CALLBACK(src, .proc/print), cooldown_per_cycle)
+
+	if(isturf(loc))
+		visible_message(span_warning("\The [src] spits out money!"))
+	else
+		visible_message(span_warning("\The [src] violently whirrs and rumbles as it tries to spit out money, but can't!"))
+		return
```
Comment (SomeguyManperson) [inline yogstation/code/game/objects/items/money_printer.dm]:
```suggestion
	if(!isturf(loc))
		visible_message(span_warning("\The [src] violently whirrs and rumbles as it tries to spit out money, but can't!"))
		return
	visible_message(span_warning("\The [src] spits out money!"))
```

File: yogstation/code/modules/uplink/uplink_item.dm [hunk 2]
```diff
@@ -143,3 +143,10 @@
 	item = /obj/item/melee/fryingpan/bananium
 	cost = 40
 	cant_discount = TRUE
+
+/datum/uplink_item/role_restricted/money_printer
+	name = "Money Printer"
+	desc = "A gigantic device that takes in nothing and puts out money. Sounds like a scam if you ask me. Don't get raided!"
+	item = /obj/item/money_printer
+	cost = 10
```
Comment (SomeguyManperson) [inline yogstation/code/modules/uplink/uplink_item.dm:151]: doubt this would be worth it especially since there's nothing to save for in cargo

Comment (YouCanBeBetterThanThat) [reply yogstation/code/modules/uplink/uplink_item.dm:151]: I have some ideas in mind if it isn't powerful enough

**Comment by AnCopper** [comment]
Yeah but, realistically, how many people would oppose literal free money for the station?

**Comment by JamieD1** [comment]
im guessing this would be used with null crates

**Comment by YouCanBeBetterThanThat** [comment]
> im guessing this would be used with null crates

Probably but nulls are quite weak unless you like bo staffs. You need a lot of money to get something good

**Comment by bruhlookatthisdood** [comment]
make it take paper to make money

**Comment by TymurShatillo** [comment]
Make it print no so much as 5000 credits. While credits are almost useless, 5000 every two minutes is still too much.

**Comment by JamieD1** [comment]
> make it take paper to make money

^^

**Comment by YouCanBeBetterThanThat** [comment]
good idea

**Comment by JamieD1** [comment]
Reopen when you work on it

**Pull request closed without merging**